app/assets/javascripts/forms.js.coffee

```diff
@@ -1,24 +1,60 @@
 $ ->
+  class Selectizer
+    constructor: (el) ->
+      @$el = $(el)
+
+    isFreeForm: ->
+      @$el.is('input')
+
+    src: ->
+      @$el.attr('data-src')
+
+    isRemote: ->
+      !!@src()
+
+    selectizeOpts: ->
+      opts = {}
+
+      if @isFreeForm()
+        opts.create = true
+        opts.maxItems = 1
+
+      if @isRemote()
+        attr = @$el.attr('data-attr')
+        opts.labelField = attr
+        opts.searchField = [attr]
+        opts.valueField = attr
+
+      opts
+
+    enable: ->
+      opts = @selectizeOpts()
+      @$el.selectize(opts)
+
+    selectizeObj: ->
+      @$el[0].selectize
+
+    onOptionsLoaded: (data) ->
+      selectize = @selectizeObj()
+      selectize.addOption(data)
+
+    loadRemoteOptions: ->
+      # TODO make sorting smarter, e.g. approvers/vendors they have used before
+      $.ajax(
+        url: @src()
+        data: {limit: 100}
```

Comment:
The API doesn't filter based on search term, that's all happening client side, right? If so, adding a limit wouldn't be a good idea, lest we only search through the first X results.